lib/rust/ensogl/core/src/debug/monitor.rs

```diff
@@ -307,25 +622,79 @@ impl Renderer {
         }
     }
 
+    fn on_label_click(&mut self, position: Vector2) {
+        let panel_index = ((position.y as f64
+            - PADDING_TOP
+            - self.user_config.outer_margin
+            - self.user_config.margin / 2.0)
+            / (self.user_config.margin + self.user_config.panel_height))
+            .floor() as usize;
+        let panel_index = panel_index.max(0).min(self.panels.len() - 1);
+        self.selected_panel = Some(panel_index);
+    }
+
+    fn on_plot_drag(&mut self, offset: Vector2) -> usize {
+        if let Some(dom) = &self.dom {
+            dom.move_selection((offset.x as f64) * SELECTION_MOVEMENT_SPEED)
+        } else {
+            0
+        }
+    }
+
+    fn set_focus_sample(&mut self, index: Option<usize>) {
```

Comment:
It seems like some additional logic would be necessary to handle `index=None`, but this function is never used that way. Maybe `index: usize`?

Reply:
Thanks for catching it!